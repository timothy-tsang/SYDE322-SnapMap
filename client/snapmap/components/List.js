import React from 'react';
import { List, ListItem, Left, Right, Button } from 'native-base';
import { View, Text, StyleSheet } from 'react-native';

export default class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    const { locations } = this.props;
    return (
      <View style={styles.list}>
        <Text style={styles.listTitle}>List of Possible Locations</Text>
        <List dataArray={locations}
          renderRow={data =>
            <ListItem onPress={() => this.props.updateMap(data.coordinate)}>
              <Left>
                <Text>{data.title}</Text>
              </Left>
              <Right>
                <Text>{data.chance}%</Text>
              </Right>
            </ListItem>
          }
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  list: {
    marginBottom: 10
  },
  listTitle: {
    textAlign: 'center',
    fontSize: 18,
    margin: 10
  },
});
