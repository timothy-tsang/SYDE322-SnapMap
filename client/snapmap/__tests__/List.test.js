import React from 'react';
import List from '../components/List';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<List />).toJSON();
  expect(tree).toMatchSnapshot();
});
