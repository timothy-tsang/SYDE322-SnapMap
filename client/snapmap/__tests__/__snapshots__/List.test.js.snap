// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 18,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    List of Possible Locations
  </Text>
  <View
    closeOnRowBeginSwipe={false}
    closeOnRowPress={true}
    closeOnScroll={true}
    dataArray={undefined}
    directionalDistanceChangeThreshold={2}
    disableLeftSwipe={false}
    disableRightSwipe={false}
    leftOpenValue={0}
    previewFirstRow={false}
    recalculateHiddenLayout={false}
    renderRow={[Function]}
    rightOpenValue={0}
    style={Object {}}
    swipeToOpenPercent={50}
    virtual={undefined}
  />
</View>
`;
